### DISCUSSION TITLE
Migration Minio to Local Storage

### DISCUSSION THREAD

**Discussion by clementthenard** [outline/outline#6201] [Self Hosting]
Hi,

I tried to migrate Minio to local storage in a self-host environnement. But I've an error when I try to download old files.

What I've done : 
- Download bucket's data
- Move bucket's data on specific folder
- Change .env variables to use local storage

What works :
- Upload file / images
- Download new file / images uploaded

What doesn't work :
- Download old files / images uploaded before migration on local storage

When I click on a file to download it, this is what it returns to me : 
- Client side :
<img width="726" alt="image" src="https://github.com/outline/outline/assets/44312491/1ee86f5d-df8a-4259-b9f3-7e98b748f7ad">

- Server side :
```
0|outline  | [Error: EISDIR: illegal operation on a directory, read] {
0|outline  |   errno: -21,
0|outline  |   code: 'EISDIR',
0|outline  |   syscall: 'read'
0|outline  | }
```

Best regards,
Clément.

**Comment by tommoor** [comment]
Interesting, you should check the `key` column in the database matches up to the path on disk. That's clearly not the case here for some reason

**Reply by clementthenard** [reply]
I don't understand because `key` column's value is `uploads/8a3d1ea1-f43e-42c3-a686-b3919e018b83/0788e1ee-4de5-4b97-ab6c-bee8dca025e4/Font Awesome Pro.zip` and `FILE_STORAGE_LOCAL_ROOT_DIR` value is `/var/lib/outline/data`. 

Obviously file exists in the directory.

**Reply by tommoor** [reply]
It could also be a permissions issue, those moved files having different permissions to newly uploaded ones

**Comment by Soleedus** [comment]
This is because you copied the MinIO bucket data directly to the local filesystem. You **must** use the mc cp utility (be sure to reference your MinIO bucket alias and not the local dirs) to copy the files otherwise Outline will attempt to access directories as if they are files -- hence the illegal operation on a directory error.

**Reply by andreyrd** [reply]
Shouldn't the documentation (quoted below) be updated to mention this?
> This is particularly useful when migrating from previously using Minio, as you can point the app at where the files were previously stored and everything will continue to work.

This is a little misleading and is exactly what I did until I found this thread.

**Reply by tommoor** [reply]
Happy to update the docs, are you able to articulate the steps that were needed? I haven't done the migration myself.

**Reply by ysmintor** [reply]
@tommoor I followed mc cp utility and assured this is the right way to migrate from Minio to Local Storage. These steps contain 

1. I used docker mc client (https://hub.docker.com/r/minio/mc/) and mount the directory to load data.
2. run docker and exec -it into the container. use mc set alias to connect the minio endpoint such as`mc alias set myminio https://myminio.example.net minioadminuser minioadminpassword`
3. use mc cp command such as `mc cp --recursive myminio/mydata/bucket01  /mount-data/`
4. change outline config from storage of s3 to local, and mount point.
5. change the owner of mount directory to 1001 such as `chown -R 1001:1001 /mount-data/`
6. docker start the outline service
